api/user/user.service.js

```diff
@@ -1,36 +1,73 @@
-'use strict';
-
-import moment from "moment";
+import moment from 'moment';
 
 const users = [];
 
 class UserService {
     constructor() {
-        let nextId = 1;
-        while (nextId <= 10) {
+        this.nextId = 0;
+
+        // Seed with some dummy data.
+        for (let i = 1; i <= 10; i++) {
+            var id = this._generateNextId();
             users.push({
-                id: nextId,
-                name: 'User Name ' + nextId,
-                email: 'user-email-name' + nextId + '@test.com',
+                id: id,
+                name: `User Name ${id}`,
+                email: `user-email-name-${id}@test.com`,
                 createdAt: '2016-12-29'
             });
-            nextId++;
         }
     }
 
     findAll(cb) {
-        let returnUsers = users;
+        let allUsers = users;
 
-        return cb(null, returnUsers)
+        return cb(null, allUsers);
     }
 
     findById(id, cb) {
-
         const user = users.find(user => user.id == id);
 
-        return cb(null, user)
+        return cb(null, user);
+    }
+
+    create(user, cb) {
+        user.id = this._generateNextId();
+        user.createdAt = moment().format('YYYY-MM-DD');
+        users.push(user);
+
+        return cb(null, user);
+    }
+
+    update(id, data, cb) {
+        // Finding the index is only for memory storage.
+        const indexOfUserToUpdate = users.findIndex((user) => { return user.id == id });
+        if (indexOfUserToUpdate === -1)
+            return cb('The user doesn\'t exists.');
```

Comment:
Also, you may want to avoid the escaping by using double quotes or template quotes.